portfolio/src/main/java/com/google/sps/comments/NewCommentServlet.java

```diff
@@ -61,8 +64,21 @@ private void addNewComment(HttpServletRequest request) {
     commentEntity.setProperty("message", commentMessage);
     commentEntity.setProperty("votes", 0);
 
+    // Determine the sentiment score of the comment
+    double sentimentScore = sentimentAnalyzer(commentMessage);
+
+    commentEntity.setProperty("sentimentScore", sentimentScore);
+
     // Put entity into datastore.
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
   }
+
+  // Analyze the overall sentiment of the comment message
+  private double sentimentAnalyzer(String commentMessage) throws IOException {
+    try (LanguageServiceClient languageService = LanguageServiceClient.create()) {
```

Comment:
What happens if this method or addNewComment throws an IOException, is it caught anywhere?

Reply:
Thanks for the comment Mark, I just updated the project and now implements error handling from the frontend, if server crashes, the page will show an error message.